### PR TITLE
Cobbduceus Part 4/X: Operating Computers Sync to Stasis Beds

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
## About The Pull Request

Operating Computers can now be built to support advanced surgeries to stasis beds. Unlike a table, these can be within LoS of the computer.

Do note it links them the same way it does a table (once when THE COMPUTER  is built).

A minor change is now the surgery tab of the computer can be accessed to sync surgeries even if a table isn't linked.

## Why It's Good For The Game

Stasis beds should be surgery hotspots, and this allows doctors to explore some of the unique pathways/surgeries without relying on a table.

Do note that some surgeries will still be better performed on the optable since the beds have a small penalty.

## Changelog
:cl: Cobby
add: Operating computers can now be built to sync with all nearby stasis beds within it's LoS and provide them with advanced surgeries.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
wtf no funny title i'm disappointed 👎 

a+ change thank you based cobby

**Comment by AdamElTablawy** [comment]
being that stasis beds don't come with op computers and people rarely build more you might as well just add one to medbay outright at roundstart near the stasis beds if you explicitly intend stasis beds to be used as surgery hotspots rather than hoping a doctor builds one every round

**Comment by Krysonism** [comment]
Power creep

Stasis is in itself very problematic as it completely eliminates any urgency and anyone brought to medbay is pretty much guaranteed to survive. This is completely antithetical to the stated goal of this rework.

There is very little reason to use a surgery table after this change.

**Comment by Dorsisdwarf** [comment]
@kryson I'm pretty sure that stasis beds have around a 10% chance to fail on each step unless you're using lots of drugs or your sterilizine sprays, which could be significant when you have surgeries like tend wounds that repeat steps over and over

**Comment by Krysonism** [comment]
> I'm pretty sure that stasis beds have around a 10% chance to fail on each step.

Yes but this is easy bypassed with free roundstart supplies.
The upside is that your patients are neigh immortal while in stasis.

Cobby has stated that he wants doctors to struggle or "spin plates" as he put it, to save critical patients. 

Stasis is diametrically opposed to that vision since it makes medical emergencies totally brain dead to handle.

If I was king I would move stasis to the techweb if not outright remove it

**Comment by Dennok** [comment]
I think about implement special resertchable stasis op table.
And you now can create it by building op table over stasis bed.

**Comment by nemvar** [comment]
While you are at it, can you fix https://github.com/tgstation/tgstation/issues/46175 by applying the same method to link the tables to the computer?

I don't like how this uses a different snowflakey way to link them. This basically screams for a unified solution.

**Comment by ExcessiveUseOfCobblestone** [comment]
I have "spinned plates" even with the stasis bed in live testing. If you're having to spin plates over on a single patient regularly you're going to have serious problems on top of the fact that healing is now pushing towards nigh-requiring singular-patient dedication.

That said I wouldn't mind these replaced with normal beds roundstart, but only after I feel like medical is comfortable enough to not need the crutch on both the player end and on the code end (I'm still just working on the base healing C2s, I haven't even looked at C3s or niche meds). That won't be for a while.

> I don't like how this uses a different snowflakey way to link them. This basically screams for a unified solution.

They use the same proc, they're just typed differently.

The table also serves a slightly different purpose since you can see patient health realtime as well as view next step for overall faster surgery (if you don't know them yet). Having it on LoS would make it more prone for multiple tables (which would be fine if I didn't eek at tgui)

In short, the idea of remotely touching tgui scares me.

**Comment by Tlaltecuhtli** [comment]
should surgery rooms have stasis tables then? like everyone is gonna move the computer round start

**Comment by ExcessiveUseOfCobblestone** [comment]
some surgeries are reliant on good probability so i'd say no. You can reorient rooms however you like though.

I need to hit the roundstart sterilizine.

**Comment by adamkad1** [comment]
> > I'm pretty sure that stasis beds have around a 10% chance to fail on each step.
> 
> Yes but this is easy bypassed with free roundstart supplies.
> The upside is that your patients are neigh immortal while in stasis.
> 
> Cobby has stated that he wants doctors to struggle or "spin plates" as he put it, to save critical patients.
> 
> Stasis is diametrically opposed to that vision since it makes medical emergencies totally brain dead to handle.
> 
> If I was king I would move stasis to the techweb if not outright remove it

yeah, if i was king i would revert cobby stuff, but neither of us are

**Comment by Detective-Google** [comment]
I feel like peridaxon as seen on cm/tgmc would be a good compromise (stops organ damage worsening and halts symptoms, letting people get to a real doctor)

**Comment by ExcessiveUseOfCobblestone** [comment]
Yes, I need to draft up a way to mitigate organ damage.
